scripts/time_plot.py

```diff
@@ -1146,8 +1146,8 @@ def UpdateCustomSignals(handlerkey,customproducts,outlierproducts,lastts):
             logger.warning("Exception while telstate set custom signals: (" + str(e) + ")", exc_info=True)
             if (handlerkey is not None):
                 send_websock_cmd('logconsole("Server exception occurred evaluating set custom signals",true,false,true)',handlerkey)
-            ingest_signals=revert_ingest_signals
             failed_update_ingest_signals_lastts=lastts
+        ingest_signals=new_ingest_signals
```

Comment:
Should this be outside the `if (changed)`? Otherwise if the set of signals is the same it won't update the timestamps?

Reply:
My intention is that this structure represents exactly what is being communicated to (and duplicated on) ingest. If it goes outside then there is some chance it is not. The timestamps indicate when the signal is added, so if it isnt added to ingest then it isnt regarded as added, and can be reconsidered on the next round of test. Stale signals are removed.

Reply:
Second thought we both have a point. Look at new commit.

Reply:
Maybe I mis-understood the purpose of the timestamps then. I thought they were tracking the last time the signal was wanted e.g. if a signal continues to be an outlier than its timestamp continues to update, so that it will linger for a minute after it ceases to be an outlier.

Reply:
Ok I'm happy with that.